nasa-gibs/worldview
# WV-3087 AERONET Layers Implementation

## Description
This adds four new layers from AERONET in the form of special vector layers. These vector layers render as vector points, but are colored by a continuous color palette based on each vector point's value. This is a unique type of layer specific to only the AERONET layers, and as such the implementation is very custom-fit to it.

## How To Test
1. `git checkout wv-3014-aeronet-aod500nm`
2. `npm ci`
3. `npm run build`
4. `npm run watch`
5. Open the 'Add Layers' modal and verify that AERONET layers can be searched for, seen in the Aerosol Optical Depth category, and are shown in the sidebar via the AERONET source.
6. Add either of the AOD AERONET layers and verify that the color palette is applied correctly to the vector points. Select a custom color palette and verify that the vector points correctly update to the custom palette. Do the same with either of the Angstrom Exponent AERONET layers as well.
7. Verify that toggling on and off the Inactive Sites vector points via the layer settings works properly and hides the inactive gray vector points correctly.
8. Add either of the near-realtime AERONET layers and verify that the timeline changes to be hourly skips instead of daily, and that transitioning across time correctly updates which vector points have an active value.
9. Verify that clicking on any of the vector points from any zoom level brings up a vector dialog, and that it contains the correct information and links for that site.
10. Verify that mousing-over a vector point correctly highlights the color palette on the side panel at the correct color, or the Inactive Sites vector color square on the side panel for inactive sites.
11. Briefly verify that other vector layers are not broken by the changes implemented here.
